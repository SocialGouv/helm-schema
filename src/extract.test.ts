import { expect, jest, test } from "@jest/globals";

import { extract } from "./index";

const tests = [
  {
    title: "YAML without comment",
    yaml: `
name: Jul
age: 42

location:
  country: FR
`,
  },
  {
    title: "YAML with comments",
    yaml: `
# @param {string} name Your name
name: Jul
# @param {number} age Your age
age: 42

location:
  # @param {string} [country] Your country
  country: FR
`,
  },
  {
    title: "YAML with nestedcomments",
    yaml: `
family:
  # @param {object} mother The mother
  mother:
    # @param {string} name The mother's name
    name:
    # @param {object} [mother] The mother's mother
    mother:
      # @param {string} [name] The mother mother's name
      name:
`,
  },
  {
    title: "YAML with sections",
    yaml: `
# @section family blablab
family:
  # @param {object} [mother] The mother
  mother:
`,
  },
  //   {
  //     title: "YAML with multiline comment",
  //     yaml: `
  // # @section family blablab
  // # more info about that crazy family
  // family:
  //   # @param {object} [mother] The mother
  //   # the mother is important in your life
  //   mother:
  // `,
  //   },
];

tests.forEach((t) => {
  test(t.title, () => {
    expect(extract(t.yaml)).toMatchSnapshot();
  });
});
