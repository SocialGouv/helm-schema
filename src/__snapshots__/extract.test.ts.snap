// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YAML with comments 1`] = `
[
  {
    "comment": {
      "description": "Your name",
      "key": "name",
      "required": true,
      "type": "string",
    },
    "key": "name",
    "value": "Jul",
  },
  {
    "key": "age",
    "value": "42",
  },
  {
    "children": [
      {
        "comment": {
          "description": "Your country",
          "key": "country",
          "required": false,
          "type": "string",
        },
        "key": "country",
        "value": "FR",
      },
    ],
    "key": "location",
    "type": "object",
  },
]
`;

exports[`YAML with nestedcomments 1`] = `
[
  {
    "children": [
      {
        "children": [
          {
            "comment": {
              "description": "The mother's name",
              "key": "name",
              "required": true,
              "type": "string",
            },
            "key": "name",
          },
          {
            "children": [
              {
                "comment": {
                  "description": "The mother mother's name",
                  "key": "name",
                  "required": false,
                  "type": "string",
                },
                "key": "name",
              },
            ],
            "comment": {
              "description": "The mother's mother",
              "key": "mother",
              "required": false,
              "type": "object",
            },
            "key": "mother",
            "type": "object",
          },
        ],
        "comment": {
          "description": "The mother",
          "key": "mother",
          "required": true,
          "type": "object",
        },
        "key": "mother",
        "type": "object",
      },
    ],
    "key": "family",
    "type": "object",
  },
]
`;

exports[`YAML with sections 1`] = `
[
  {
    "children": [
      {
        "comment": {
          "description": "The mother",
          "key": "mother",
          "required": false,
          "type": "object",
        },
        "key": "mother",
      },
    ],
    "comment": {
      "description": "family blablab",
    },
    "key": "family",
    "type": "object",
  },
]
`;

exports[`YAML without comment 1`] = `
[
  {
    "key": "name",
    "value": "Jul",
  },
  {
    "key": "age",
    "value": "42",
  },
  {
    "children": [
      {
        "key": "country",
        "value": "FR",
      },
    ],
    "key": "location",
    "type": "object",
  },
]
`;
